opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp

```diff
@@ -406,6 +406,13 @@ namespace Opm
                          SimulationDataContainer& simulatorstate,
                          WellStateFullyImplicitBlackoil& wellstate)
     {
+        std::map<std::string, UnitSystem::measure> solution_keys {{"PRESSURE" , UnitSystem::measure::pressure},
+                                                                  {"SWAT" , UnitSystem::measure::identity},
+                                                                  {"SGAS" , UnitSystem::measure::identity},
+                                                                  {"TEMP" , UnitSystem::measure::temperature},
+                                                                  {"RS" , UnitSystem::measure::identity},
```

Comment:
The measure used to convert from on-disk-units to SI units, the conversion takes place in opm-output, so that opm-simulator will only see SI data. As you describe it it is clearly wrong; however I fear you are pointing to something which is wrong also in output direction (I guess that might be why it has worked ...). I will look into it.